padertorch/data/fragmenter.py

```diff
@@ -107,3 +107,88 @@ def fragment(key, x):
             fragment = deflatten(fragment)
             fragments.append(fragment)
         return fragments
+
+
+
+def get_segment_offsets(num_samples, length, shift=None,
+                        offset_mode='start', num_segments=None):
+    """
+
+    Args:
+        num_samples:
+        length:
+        shift:
+        offset_mode:
+        num_segments:
+
+    Returns:
+    >>> get_segment_offsets(25, 10, None, offset_mode='start')
+    array([ 0, 10])
+    >>> get_segment_offsets(25, 10, 3, offset_mode='start')
+    array([ 0,  3,  6,  9, 12, 15])
+    >>> get_segment_offsets(24, 10, 3, offset_mode='start')
+    array([ 0,  3,  6,  9, 12])
+    >>> get_segment_offsets(24, 10, 3, offset_mode='end')
+    array([ 2,  5,  8, 11, 14])
+    >>> get_segment_offsets(24, 10, 3, offset_mode='center')
+    array([ 1,  4,  7, 10, 13])
+    >>> get_segment_offsets(24, 10, 3, offset_mode='center', num_segments=2)
+    array([1, 4])
+    >>> get_segment_offsets(24, 10, 3, offset_mode='center', num_segments=1)
+    array([1])
+    """
+    assert num_samples >= length, (num_samples, length)
+    if shift is None:
+        shift = length
+    assert shift > 0, shift
+
+    if offset_mode == 'start':
+        start = 0
+    elif offset_mode == 'random':
+        start = np.random.randint(num_samples - length)
+    elif offset_mode in ['end', 'center']:
+        remainder = (num_samples - length) % shift
+        if offset_mode == 'end':
+            start = remainder
+        else:
+            start = remainder // 2
+    else:
+        raise ValueError('Unknown offset mode', offset_mode)
+    offset = np.arange(start, num_samples - length + 1, shift)
+
+    if num_segments:
+        max_segments = (num_samples - length) // shift + 1
+        assert num_segments <= max_segments, (num_segments, max_segments)
+        offset = offset[:num_segments]
+    return offset
+
+
+def segment(x, length, offset=None, shift=None, axis=-1,
+            offset_mode='start', num_segments=None):
+    """
+
+    Args:
+        x:
+        segment_length:
+        step:
+        axis:
+        offset_mode:
+        num_segments:
+
+    Returns:
+
+    >>> segment(np.arange(0, 15), 10, None, 3, -1)
+    array([[ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9],
+           [ 3,  4,  5,  6,  7,  8,  9, 10, 11, 12]])
+    """
+
+    if offset is None:
+        offset = get_segment_offsets(x.shape[axis], length, shift,
+                                      offset_mode, num_segments)
```

Comment:
Is it necessary to call get_segment_offsets here or would it be enough to move the "offset_mode" part to a separate function and use that function here?